backend/src/main/java/ai/verta/modeldb/dataset/DatasetDAORdbImpl.java

```diff
@@ -154,31 +154,19 @@ private void createWorkspaceRoleBinding(
       String datasetId,
       DatasetVisibility datasetVisibility) {
     if (workspaceId != null && !workspaceId.isEmpty()) {
-      Role datasetAdmin = roleService.getRoleByName(ModelDBConstants.ROLE_DATASET_ADMIN, null);
+      roleService.createWorkspaceRoleBinding(workspaceId, workspaceType, datasetId, ModelDBConstants.ROLE_DATASET_ADMIN, ModelDBServiceResourceTypes.DATASET);
       Role datasetRead = roleService.getRoleByName(ModelDBConstants.ROLE_DATASET_READ_ONLY, null);
       switch (workspaceType) {
         case ORGANIZATION:
-          Organization org = (Organization) roleService.getOrgById(workspaceId);
-          roleService.createRoleBinding(
-              datasetAdmin,
-              new CollaboratorUser(authService, org.getOwnerId()),
-              datasetId,
-              ModelDBServiceResourceTypes.DATASET);
           if (datasetVisibility.equals(DatasetVisibility.ORG_SCOPED_PUBLIC)) {
```

Comment:
My mistake if this was missed earlier but let us have ORG_SCOPED_PUBLIC visibility for repo and then move this to roleService. createWorkspaceRoleBinding

Reply:
https://github.com/VertaAI/modeldb/pull/479